File: sdk/fsharp/src/HiTechCloudSDK/ServiceStatus.fs
namespace HiTechCloud.SDK

/// Service Status: list and manage service statuses
type ServiceStatusResource(httpClient: HttpClient) =

    /// GET /api/statuses - Returns a list of all statuses
    member this.List() =
        async {
            return! httpClient.getAsync("/api/statuses")
        }

    /// PUT /api/statuses/{id} - Returns details of status
    member this.Get() =
        async {
            return! httpClient.putAsync("/api/statuses/{id}")
        }

File: sdk/clojure/src/hitechcloud/sdk/service_status.clj
(ns hitechcloud.sdk.service_status
  (:require [hitechcloud.sdk.http-client :as http]))

;; Service Status: list and manage service statuses

(defn list "GET /api/statuses - Returns a list of all statuses" [client & [params]] (http-get client "/api/statuses" params))

(defn get "PUT /api/statuses/{id} - Returns details of status" [client & [params]] (http-put client "/api/statuses/{id}" params))

